ui/src/app/streams/stream-deploy/stream-deploy.component.ts

```diff
@@ -88,6 +88,12 @@ export class StreamDeployComponent implements OnInit, OnDestroy {
     this.route.params.subscribe((params: Params) => {
       this.config$ = this.streamDeployService
         .config(params.id)
+        .pipe(catchError(val => {
+          if (val === `Could not find stream definition named ${params.id}`) {
```

Comment:
Not sure - I like this.

Maybe we should defer to `1.5` and handle/solve this more appropriately. Our current error handler is somewhat limited - returning a simple string has not been the best thing _since sliced bread_. We should either add a generic `NotFoundException` that can be caught by component code or we treat a `404` status specifically, causing business methods to return `undefined` in that case.